### ISSUE TITLE
Feature: Run NCA simultaneously over multiple analytes

### ISSUE THREAD

**Issue by Gero1999** [pharmaverse/aNCA#3]
**Issue description**
The current used PKNCA package allows the analysis to be run on multiple analytes simultaneously, however our current implementation only provides input selections for one analyte at a time. This can be changed in order to fully satisfy the user's needs. 

- [ ] User should be able to multiple choice in main parameters as now (`ANALYTE`, `DOSNO`) and new ones (`PCSPEC`)
- [ ] Multiple analytes should be able to be run, and slope selector should keep working well
- [ ] Different analytes with different ROUTE, MATRIX...

**Notes**
This feature can be of great interest to enable other future enhancements to the app; i.e, the customization of analyte ratios

**Comment by billdenney** [comment]
Related to #125

This can be done as either a grouping variable or as an analyte using PKNCA. Over time, I think that most people have shifted from using the analyte formula notation to using `PCTEST` as a grouping variable.

**Comment by Gero1999** [comment]
Yes, indeed this challenge is more about readapting our current code that any difficulty with PKNCA. Regarding this topic, looks these formulas ok? 

Conc: `AVAL ~ TIME | STUDYID + PCSPEC + DRUG + USUBJID / ANALYTE`
Dose: `DOSEA ~ TIME | STUDYID + PCSPEC + DRUG + USUBJID + DOSNO`

I know that `DOSNO` is not strictly needed, but for the intervals we want that they all start from 0 using as `TIME` the time since last dose. We will make sure to duplicate all predose records so if needed PKNCA can take them into consideration. Apart from that, we did not think of other potential issues

**Comment by billdenney** [comment]
The way that PKNCA works, the dose grouping must be a subset of the conc groups. So, I believe that the suggested groups would fail when combining at the `PKNCAdata()` step because `DOSENO` is not in the conc groups (if not, that's a bug that should be fixed in PKNCA).

Out of curiosity, why are you starting all intervals at `TIME` 0? PKNCA can handle that for you, and the impute methods can put all of the predose records in for you (using the `impute` functionality) so that you don't need to do outside data management. If it isn't doing everything you need here, that is functionality that should be added to `impute` functionality within PKNCA. There is a goal that (almost) no data management is required before putting SDTM data into PKNCA.

**Comment by Gero1999** [comment]
Ups yes you are right I missed writting also `DOSNO` in the concentration groups formula, yes that for sure is what makes sense.

The thing is that our users have communicated us that they would like that the results start with respect to the last dose time instead of the accumulated time. Of course, we could still do this as a postprocessing step to adjust `start` and `end`, but not sure if that would make things more complicated than just duplicating predoses... 

I would also like to ask you about how the nominal time can be used (if so) with the checking of the direct results or if there is any ongoing plan about this. How are NAs handled for this column as well?

**Comment by billdenney** [comment]
Parameters like `tmax` and others that are calculated are always calculated relative to the start of the interval. That usually covers users' requests there. If you separate the data with `DOSENO` groups, it will break PKNCA's internal imputation methods because PKNCA will not look outside of the current group for imputing values.

How are you wanting to use nominal times? The time used for calculations is whatever is given in the formula.  You can give either nominal or actual times, whichever you want to use for calculations; usually you want to use actual times for calculations. I have internal plotting routines that look at nominal times when creating group plots (e.g. mean or median by treatment).

`NA` values in time are an error. `NA` values in concentrations are excluded from calculations. `NA` values in grouping variables are treated as any other group.
